jest-rs/src/main/java/io/searchbox/rs/client/http/JestRsClient.java

```diff
@@ -0,0 +1,153 @@
+package io.searchbox.rs.client.http;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.Map.Entry;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.Invocation.Builder;
+import javax.ws.rs.client.InvocationCallback;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import com.google.gson.Gson;
+
+import io.searchbox.action.Action;
+import io.searchbox.client.AbstractJestClient;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.client.JestResultHandler;
+import io.searchbox.core.Explain;
+
+/**
+ * @author Dogukan Sonmez
+ * @author cihat keser
+ */
+public class JestRsClient extends AbstractJestClient implements JestClient {
+
+    private final static Logger log = Logger.getLogger(JestRsClient.class.getName());
+    // private final static HttpEntity DEFAULT_OK_RESPONSE = EntityBuilder.create().setText("{\"ok\" : true, \"found\" : true}").build();
+    // private final static HttpEntity DEFAULT_NOK_RESPONSE = EntityBuilder.create().setText("{\"ok\" : false, \"found\" : false}").build();
```

Comment:
Checking it again, you're right; none of the tests fail if that logic is removed. I believe it was put in place because parsing logic in action classes did not null check before but they do now. This can be removed safely, I'll also remove it on other clients too.